interface/main/calendar/index.php

```diff
@@ -97,7 +106,8 @@
 
 // Simplifying by just using request variable instead of checking for both post and get - KHY
 if (isset($_REQUEST['viewtype'])) {
-    $_SESSION['viewtype'] = $_REQUEST['viewtype'];
+    $viewtype=filter_var( $_REQUEST['viewtype'],FILTER_SANITIZE_STRING);
+    $_SESSION['viewtype'] = $viewtype;
```

Comment:
On all 3 above, again. much better to get this downstream where it is echoed to html.